cmd/ingress-operator/start.go

```diff
@@ -157,14 +170,14 @@ func start() error {
 					close(stop)
 					return
 				}
-				latest, err := ioutil.ReadFile(defaultTrustedCABundle)
+				latest, err := ioutil.ReadFile(shutdownFile)
```

Comment:
We could use the event to make the logic more general (which would accommodate my suggestion to take a slice of file names):
```go
			case event, ok := <-watcher.Events:
				// ...
				latest, err := ioutil.ReadFile(event.Name)
```
We would also need to make sure to canonicalize path names, and we would need a map to track multiple files' contents:
```go
	orig := map[string][]byte{}
	for _, path := range shutdownFiles {
		realPath, err := filepath.EvalSymlinks(path)
		if err != nil {
			// ...
		}
		if err := watcher.Add(realPath); err != nil {
			// ...
		}
		orig[realPath], err = ioutil.ReadFile(realPath)
		// ...
	}
	// ...
	go func() {
		for {
			select {
				case event, ok := <-watcher.Events:
				// ...
				realPath, err := filepath.EvalSymlinks(event.Name)
				// ...
				if !bytes.Equal(orig[realPath], latest) {
					// ...
```
So it would be a nicer API IMO but a messier implementation; what do you think?

Reply:
It sounds nice but at the end of the day the only reason I'm making any change here at all is to facilitate local development, so I'm starting to get worried I'm gold plating something that doesn't really matter much at all

Reply:
> It sounds nice but at the end of the day the only reason I'm making any change here at all is to facilitate local development, so I'm starting to get worried I'm gold plating something that doesn't really matter much at all

If it's for local dev only would an environment variable suffice (and take precedence)?

Reply:
Options are nicer and basically free for us to use